### DISCUSSION TITLE
Implementing Jellyroll Model

### DISCUSSION THREAD

**Discussion by jouslin** [pybamm-team/PyBaMM#4237] [Q&A]
Hi all,
I've previously set up a _lithium_ion.DFN()_ model with a specific set of parameters (many parameters from _Chen 2020_), a _lumped thermal model_, and a _pouch geometry cell_. However, now I want to change the lumped thermal model and the pouch geometry to a **jellyroll geometry** and **1D spatial dependence for temperature** (along the radial direction of the cylinder). I have seen that there is helper code to implement this type of geometry.

````
N = pybamm.Parameter("Number of winds")
r0 = pybamm.Parameter("Inner radius")
eps = (1 - r0) / N  # ratio of sandwich thickness to cell radius
delta = pybamm.Parameter("Current collector thickness")
delta_p = delta  # assume same thickness
delta_n = delta  # assume same thickness
l = 1 / 2 - delta_p - delta_n  # active material thickness
sigma_p = pybamm.Parameter("Positive current collector conductivity [S.m-1]")
sigma_n = pybamm.Parameter("Negative current collector conductivity [S.m-1]")
sigma_a = pybamm.Parameter("Active material conductivity")


r = pybamm.SpatialVariable("radius", domain="cell", coord_sys="cylindrical polar")
geometry = {"cell": {r: {"min": r0, "max": 1}}}


model = pybamm.BaseModel()
phi_p = pybamm.Variable("Positive potential", domain="cell")
phi_n = pybamm.Variable("Negative potential", domain="cell")

A_p = (2 * sigma_a / eps**4 / l) / (delta_p * sigma_p / 2 / pi**2)
A_n = (2 * sigma_a / eps**4 / l) / (delta_n * sigma_n / 2 / pi**2)
model.algebraic = {
    phi_p: pybamm.div((1 / r**2) * pybamm.grad(phi_p)) + A_p * (phi_n - phi_p),
    phi_n: pybamm.div((1 / r**2) * pybamm.grad(phi_n)) - A_n * (phi_n - phi_p),
}

model.boundary_conditions = {
    phi_p: {
        "left": (0, "Neumann"),
        "right": (1, "Dirichlet"),
    },
    phi_n: {
        "left": (0, "Dirichlet"),
        "right": (0, "Neumann"),
    },
}
model.initial_conditions = {phi_p: 1, phi_n: 0}  # initial guess for solver
model.variables = {"Negative potential": phi_n, "Positive potential": phi_p}

parameters.process_geometry(geometry)
parameters.process_model(model)

submesh_types = {"cell": pybamm.Uniform1DSubMesh}
var_pts = {r: 100}
mesh = pybamm.Mesh(geometry, submesh_types, var_pts)

spatial_methods = {"cell": pybamm.FiniteVolume()}

disc = pybamm.Discretisation(mesh, spatial_methods)
disc.process_model(model)
```` 

Nonetheless, it uses a completely new model. I wanted to know if it's possible to add Jellyroll equations, parameters, and mesh directly into my existing model without creating another one. 
I've already tried to add them, but it didn't work. Additionally, I attempted to install JellyBaMM, but it requires a previous version of PyBaMM and I'm currently using the latest version, so they are not compatible.

Thanks for your help :)

**Comment by procrastinator10125** [comment]
You can install JellyBaMM with previous version of python, it does not have much dependencies on PyBaMM, but issues with other libraries. Try creating a virtual environment and installing JellyBaMM with it, in an older version of python, preferably <3.10
